### PR TITLE
Implement linux target

### PR DESCRIPTION
**Pull request by chippmann**
This implements `linuxX64` as a target with the corresponding linux specific actual implementations.
Primarily this was implemented to hopefully pave the way for https://github.com/cashapp/sqldelight/issues/1835.

I did some minor testing with a local build of `SQLDelight` and so far everything seems to work.
~~I did not figure out how your test setup with that arbitrary script could work and based on @kpgalligan's [comment](https://github.com/cashapp/sqldelight/issues/1835#issuecomment-656298033) on the above mentioned issue I did not spend too much time on it as it's probably rewritten anyways? But if you require it I'm of course willing to spend some more time to figure that one out as well.~~ Tests are run after latest rebase.

### CODE REVIEW

File: sqliter-driver/src/linuxX64Main/kotlin/co/touchlab/sqliter/DatabaseFileContext.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    actual fun deleteDatabase(name: String, basePath: String?) {
+        deleteDatabaseFile(databaseFile(name, basePath))
+    }
+
+    actual fun databasePath(databaseName: String, datapathPath: String?): String {
+        return databaseFile(databaseName, datapathPath).path
+    }
+
+    internal fun databaseDirPath(): String {
+        return getHomeDirPath() ?: memScoped {
```
Comment (chippmann) [inline sqliter-driver/src/linuxX64Main/kotlin/co/touchlab/sqliter/DatabaseFileContext.kt:23]: Actually here I'm not sure where the fallback path should lead to?
I just defined the `home` dir but of course this is suboptimal. But another arbitrary fallback dir is also IMO. But you did something similar on windows as well and one should configure the path anyways in most cases.
Maybe you have a preference where the fallback path should be? Or is `home` fine for you?

**Comment by chippmann** [comment]
Any change this could get looked at or is there no interest in it?

**Comment by carlonzo** [comment]
@kpgalligan any chance we can have this reviewed/merged? would be great to support Linux targets on SQLDelight

**Comment by kpgalligan** [comment]
I tried running tests and got a failure with the following:

```
e: /home/runner/.konan/dependencies/x86_64-unknown-linux-gnu-gcc-8.3.0-glibc-2.19-kernel-4.9-2/x86_64-unknown-linux-gnu/bin/ld.gold invocation reported errors

> Task :sqliter-driver:linkDebugTestLinuxX64 FAILED
The /home/runner/.konan/dependencies/x86_64-unknown-linux-gnu-gcc-8.3.0-glibc-2.19-kernel-4.9-2/x86_64-unknown-linux-gnu/bin/ld.gold command returned non-zero exit code: 1.
output:
/usr/lib/x86_64-linux-gnu/libsqlite3.so: error: undefined reference to 'log', version 'GLIBC_2.29'
/usr/lib/x86_64-linux-gnu/libsqlite3.so: error: undefined reference to 'fcntl64', version 'GLIBC_2.28'
```

Can provide more log if needed.

**Comment by kpgalligan** [comment]
To clarify above, I made another branch with your changes and pushed to get the linux tests running (rather than set up a local virtual machine). The error above is from the test log. I have roughly zero linux dev experience that wasn't on the JVM, so it would probably take me a bit to sort out that issue. If you have any thoughts, please share.

**Comment by audkar** [comment]
Were libsqlite or lib binaries prebuilt on other CI instance/machine? Probably some binary was built on OS which had newer glibc. To ensure compatibility need to ensure that everything is built on older OS with older glibc

**Comment by chippmann** [comment]
Hmm to me it seems the GLIB_C version mismatches. Probably the konan toolchain uses a different one now. It probably broke with my last rebase as there the kotlin version changed to 1.5.x IIRC and i didn't notice it.
I'll look into it this weekend.

**Comment by chippmann** [comment]
> Were libsqlite or lib binaries prebuilt on other CI instance/machine? Probably some binary was built on OS which had newer glibc. To ensure compatibility need to ensure that everything is built on older OS with older glibc

The problem with this is that konan (AFAIK) bundles fixed versions: https://github.com/JetBrains/kotlin-native/blob/1d2f55ce23d050cad2f3611a61261bf6a509a36f/konan/konan.properties#L411
Not sure if one can change the version used.

**Comment by chippmann** [comment]
It seems @audkar was right. Simply using ubuntu 18.04 to build it was enough.
I updated the workflows accordingly.
@kpgalligan Could you either approve the workflow in this PR or trigger it yourself to test whether this also works in CI?

**Comment by chippmann** [comment]
> Just tried the same thing and it seems like it's working: https://github.com/touchlab/SQLiter/actions/runs/968436878. I'll let that finish then approve.

Very nice!
Thx a lot!

**Comment by kpgalligan** [comment]
Thanks for the PR. Sorry for the delay. Going to see about getting this into SQLDelight.

**Comment by chippmann** [comment]
> Thanks for the PR. Sorry for the delay. Going to see about getting this into SQLDelight.

No problem.
Once a release is done, I'll create a PR in the SQLDelight repo.

**Comment by chippmann** [comment]
@kpgalligan it seems i forgot to update one `linux-latest` for the deploy: https://github.com/touchlab/SQLiter/blob/5432b6c0a95ed0cb13ae459af14a86de83aaa4c1/.github/workflows/deploy.yml#L62
Sorry about that.
Do you update it?

**Comment by kpgalligan** [comment]
I'll update...

**Pull request merged**